Cast-align warning in OS_TaskGetId_Impl for vxworks

**Is your feature request related to a problem? Please describe.**
```
/home/jhageman/cFS-GitHub/osal/src/os/vxworks/src/os-impl-tasks.c: In function ‘OS_TaskGetId_Impl’:
/home/jhageman/cFS-GitHub/osal/src/os/vxworks/src/os-impl-tasks.c:411:12: warning: cast increases required alignment of target type [-Wcast-align]
  411 |     lrec = (OS_impl_task_internal_record_t *)taskTcb(taskIdSelf());
```

Recreate by building on Ubuntu 20.04, set `cast-align=strict` in `sample_defs/arch_build_custom.cmake` and prep with `ENABLE_UNIT_TESTS=true`

**Describe the solution you'd like**
Resolve warnings.

**Describe alternatives you've considered**
None

**Additional context**
Observed on Ubuntu 20.04 w/ cast-align=strict

**Requester Info**
Jacob Hageman - NASA/GSFC

--- a/src/os/vxworks/src/os-impl-tasks.c
+++ b/src/os/vxworks/src/os-impl-tasks.c
@@ -400,16 +400,16 @@ int32 OS_TaskRegister_Impl(osal_id_t global_task_id)
  *-----------------------------------------------------------------*/
 osal_id_t OS_TaskGetId_Impl(void)
 {
-    OS_impl_task_internal_record_t *lrec;
-    size_t                          idx;
-    osal_id_t                       id;
+    void     *lrec;
+    size_t    idx;
+    osal_id_t id;
 
     id   = OS_OBJECT_ID_UNDEFINED;
-    lrec = (OS_impl_task_internal_record_t *)taskTcb(taskIdSelf());
+    lrec = taskTcb(taskIdSelf());
 
     if (lrec != NULL)
     {
-        idx = lrec - &OS_impl_task_table[0];
+        idx = (OS_impl_task_internal_record_t *)lrec - &OS_impl_task_table[0];
         if (idx < OS_MAX_TASKS)
         {
             id = OS_global_task_table[idx].active_id;

